src/os/bluestore/BlueStore.cc

```diff
@@ -11601,6 +11601,8 @@ int BlueStore::set_collection_opts(
     return -ENOENT;
   std::unique_lock l{c->lock};
   c->pool_opts = opts;
+  c->segment_size = c->pool_opts.value_or(
```

Comment:
Do we really need this to be a per-pool option? The latter implies this is user-configurable, What would inspire users to tune that? Any metrics, recommendations, etc?

Reply:
Segmentation brings hard data separation. It limits the scan range for compression / recompression and for onode sharding.
Having larger segments enables larger compressed blob and reduced wastage.
Having smaller segments guarantees finer grain for sharding that does not create spanning blobs.

Do we really need this to be a per-pool option? Maybe not, but we already have options for min/max compressed blob size, minimal compression gain and compression mode. It just follows the pattern.

Reply:
Maybe we should leave this out, until the customers request the pool option.

Reply:
In some relevant sense one could argue that there is a relation between
- onode_segment_size 
and
- max_compressed_blob_size
But how to explain why non-compressed onodes are using settings for compressed case?

The logic of setting perfect `onode_segment_size` is very convoluted. It impacts size of onode shards. And we know we want our shards not too big, but at the same time we do not want to have too many of them.
Could we devise some formula that takes into account:
- onode checksum type
- onode checksum size
- typical encoding sizes of allocated extents
But it is difficult to predict compression ratio (and this determines amount of checksum data)